rest-api-spec/src/main/resources/rest-api-spec/test/get/100_mix_typeless_typeful.yml

```diff
@@ -0,0 +1,46 @@
+---
+"GET with typeless API on an index that has types":
+
+ - skip:
+      version: " - 6.99.99"
+      reason:  Typeless APIs were introduced in 7.0.0
+
+ - do:
+      indices.create: # not using include_type_name: false on purpose
+          index: index
+          body:
+              mappings:
+                  not_doc:
+                      properties:
+                          foo:
+                              type: "keyword"
+
+ - do:
+      index:
+          index:  index
+          type:   not_doc
+          id:     1
+          body:   { foo: bar }
+
+ - do:
+      catch:  missing
+      get:
+          index:  index
+          type:   some_random_type
+          id:     1
+
+ - match:   { _index:   "index" }
+ - match:   { _type:    "some_random_type" }
+ - match:   { _id:      "1"}
+ - match:   { found:      false}
+
+ - do:
+      get:
+          index:  index
+          id:     1
+
+ - match:   { _index:   "index" }
+ - match:   { _type:    "_doc" }
+ - match:   { _id:      "1"}
+ - match:   { _version: 1}
+ - match:   { _source: { foo: bar }}
```

Comment:
do we also need to check whether the doc comes back when specifying not_doc as a type?

Reply:
I didn't add a test for it because this is already tested by `get/11_basic_with_types.yml`?